### PR TITLE
show more button beneath mutation table

### PR DESCRIPTION
**Pull request by raerose01**
Add a button beneath mutation table to show additional mutations in patient view.

Changes proposed in this pull request:
- additional button beneath mutation table

showmorebutton

### CODE REVIEW

File: src/shared/components/paginationControls/ButtonControls.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
```
Comment (inodb) [inline src/shared/components/paginationControls/ButtonControls.tsx]: Only comment here is maybe we can change the name of `ButtonControls` to `ShowMoreButton`

File: src/shared/components/paginationControls/ButtonControls.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+import {Button, ButtonGroup, FormGroup, FormControl} from 'react-bootstrap';
```
Comment (inodb) [inline src/shared/components/paginationControls/ButtonControls.tsx]: seems like there are some unused imports here?

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx [hunk 3]
```diff
@@ -546,6 +551,15 @@ export default class LazyMobXTable<T> extends React.Component<LazyMobXTableProps
         }
     }
 
+    private getButtonControls() {
```
Comment (inodb) [inline src/shared/components/lazyMobXTable/LazyMobXTable.tsx]: maybe rename to `getShowMoreButtonControls` or `getShowMoreButton` and component accordingly

File: src/shared/components/paginationControls/ButtonControls.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    constructor(props:IButtonControlsProps) {
+        super(props);
+        this.handleChangeItemsPerPage = this.handleChangeItemsPerPage.bind(this);
+        this.counter=0;
+        this.showMoreItems=[10,25,50,100,-1]
+    }
+
+    handleChangeItemsPerPage() {
+
+        if(this.props.onChangeItemsPerPage){
```
Comment (inodb) [inline src/shared/components/paginationControls/ButtonControls.tsx]: styling, might want to add space after `if`

File: src/shared/components/paginationControls/ButtonControls.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            if (this.props.itemsPerPage) {
+                this.counter = this.showMoreItems.indexOf(this.props.itemsPerPage) + 1
+                if(this.counter < this.showMoreItems.length){
```
Comment (inodb) [inline src/shared/components/paginationControls/ButtonControls.tsx]: same here styling

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx [hunk 6]
```diff
@@ -595,6 +617,9 @@ export default class LazyMobXTable<T> extends React.Component<LazyMobXTableProps
                 <Observer>
                     {this.getTable}
                 </Observer>
+                <Observer>
+                    {this.showMore}
```
Comment (inodb) [inline src/shared/components/lazyMobXTable/LazyMobXTable.tsx]: Seems like this is wrapped in two `Observer` tags, I don't think you really need the `showMore` function.

**Comment by jjgao** [comment]
Maybe we can make it similar to Google Scholar's button and merge the prev/next button together.

image

https://scholar.google.com/citations?user=a5JPZ6MAAAAJ&hl=en

**Comment by inodb** [comment]
We made some updates to the design: #499 . Thanks was able to reuse some of these ideas!

**Pull request closed without merging**